chapter9/phone-agent/speech.py

```
@@ -0,0 +1,307 @@
+        if not cleaned:
+            raise ValueError("TTS text must not be empty")
+        started = time.monotonic()
+        with tempfile.TemporaryDirectory(prefix="exp9-2-tts-") as directory:
+            directory_path = Path(directory)
+            source = directory_path / ("speech.aiff" if self.kind == "say" else "speech.wav")
+            target = directory_path / "speech.wav"
+            if self.kind == "say":
+                command = [self.engine, "-v", self.voice, "-o", str(source), cleaned]
+            else:
+                command = [self.engine, "-v", self.voice, "-w", str(source), cleaned]
+            subprocess.run(command, check=True, capture_output=True, timeout=90)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)